docs/ANDROID_CROSS_ON_M1_MAC.md

```diff
@@ -0,0 +1,20 @@
+## Cross-compiling MM for Android on M1 Mac
+
+1. Ensure that your terminal is added to `Developer tools` in MacOS Security & Privacy settings.
+2. The cross-compilation requires Android NDK version 21. Custom brew cask file is located at the root of this repo.
+3. Install android-ndk by `brew install --cask android-ndk.rb` or other preferred way.
+4. Add Android targets via rustup
+```shell
+rustup target add armv7-linux-androideabi
+rustup target add aarch64-linux-android
+rustup target add i686-linux-android
+rustup target add x86_64-linux-android
+```
+5. The following build commands presume that NDK was installed using brew. If it was installed other way, you have to change PATH ENV accordingly.
+```shell
+export PATH=$PATH:/opt/homebrew/share/android-ndk/toolchains/llvm/prebuilt/darwin-x86_64/bin
+CARGO_TARGET_ARMV7_LINUX_ANDROIDEABI_LINKER=armv7a-linux-androideabi21-clang CC_armv7_linux_androideabi=armv7a-linux-androideabi21-clang AR_armv7_linux_androideabi=llvm-ar CXX_armv7_linux_androideabi=armv7a-linux-androideabi21-clang++ ANDROID_NDK_HOME="/opt/homebrew/share/android-ndk" cargo build --target=armv7-linux-androideabi --release --lib
+CARGO_TARGET_AARCH64_LINUX_ANDROID_LINKER=aarch64-linux-android21-clang CC_aarch64_linux_android=aarch64-linux-android21-clang AR_aarch64_linux_android=llvm-ar CXX_aarch64_linux_android=aarch64-linux-android21-clang++ ANDROID_NDK_HOME="/opt/homebrew/share/android-ndk" cargo build --target=aarch64-linux-android --release --lib
+CARGO_TARGET_I686_LINUX_ANDROID_LINKER=i686-linux-android21-clang CC_i686_linux_android=i686-linux-android21-clang AR_i686_linux_android=llvm-ar CXX_i686_linux_android=i686-linux-android21-clang++ ANDROID_NDK_HOME="/opt/homebrew/share/android-ndk" cargo build --target=i686-linux-android --release --lib
+CARGO_TARGET_X86_64_LINUX_ANDROID_LINKER=x86_64-linux-android21-clang CC_x86_64_linux_android=x86_64-linux-android21-clang CXX_x86_64_linux_android=x86_64-linux-android21-clang++ AR_x86_64_linux_android=llvm-ar ANDROID_NDK_HOME="/opt/homebrew/share/android-ndk" cargo build --target=x86_64-linux-android --release --lib
```

Comment:
What do you think about formatting these commands by combining env variables and the commands into groups?
For example,
```bash
export PATH=$PATH:/opt/homebrew/share/android-ndk/toolchains/llvm/prebuilt/darwin-x86_64/bin
export CARGO_TARGET_ARMV7_LINUX_ANDROIDEABI_LINKER=armv7a-linux-androideabi21-clang
export CC_armv7_linux_androideabi=armv7a-linux-androideabi21-clang
export AR_armv7_linux_androideabi=llvm-ar
export CXX_armv7_linux_androideabi=armv7a-linux-androideabi21-clang++
export ANDROID_NDK_HOME="/opt/homebrew/share/android-ndk"

cargo build --target=armv7-linux-androideabi --release --lib
```

Reply:
I even didn't notice that there are several build commands until I tried to compile it locally :grin:

Reply:
Considering tricky environment handling behavior as here https://github.com/KomodoPlatform/atomicDEX-API/issues/1176#issuecomment-1000784537, I will have to recheck that every such build command will actually work. And I prefer to not do this :slightly_smiling_face: I will leave some comments to separate commands of each target, so it should become more clear.

Reply:
@sergeyboyko0791 I hope it looks better now: https://github.com/KomodoPlatform/atomicDEX-API/blob/mm2.1-cross/docs/ANDROID_CROSS_ON_M1_MAC.md